node/src/blockchain/blockchain_bridge.rs

```diff
@@ -720,7 +659,157 @@ mod tests {
     }
 
     #[test]
-    fn handle_report_account_payable_manages_gas_price_error() {
+    fn handle_report_accounts_payable_transacts_an_eleventh_hour_error_down_to_accountant() {
+        let system = System::new(
+            "handle_report_accounts_payable_transacts_an_eleventh_hour_error_down_to_accountant",
+        );
+        let (accountant, _, accountant_recording_arc) = make_recorder();
+        let hash = make_tx_hash(222);
+        let wallet_account = make_wallet("blah");
+        let expected_error_msg = "We were so close but we stumbled and smashed our face against \
+         the ground just a moment after the signing";
+        let expected_error = Err(BlockchainError::PayableTransactionFailed {
+            msg: expected_error_msg.to_string(),
+            signed_and_saved_txs_opt: Some(vec![hash]),
+        });
+        let blockchain_interface_mock = BlockchainInterfaceMock::default()
+            .get_transaction_count_result(Ok(U256::from(1u64)))
+            .send_payables_within_batch_result(expected_error.clone());
+        let persistent_configuration_mock =
+            PersistentConfigurationMock::default().gas_price_result(Ok(123));
+        let consuming_wallet = make_paying_wallet(b"somewallet");
+        let subject = BlockchainBridge::new(
+            Box::new(blockchain_interface_mock),
+            Box::new(persistent_configuration_mock),
+            false,
+            Some(consuming_wallet),
+        );
+        let addr = subject.start();
+        let subject_subs = BlockchainBridge::make_subs_from(&addr);
+        let peer_actors = peer_actors_builder().accountant(accountant).build();
+        let accounts = vec![PayableAccount {
+            wallet: wallet_account,
+            balance_wei: 111_420_204,
+            last_paid_timestamp: from_time_t(150_000_000),
+            pending_payable_opt: None,
+        }];
+        send_bind_message!(subject_subs, peer_actors);
+
+        let _ = addr
+            .try_send(ReportAccountsPayable {
+                accounts,
+                response_skeleton_opt: Some(ResponseSkeleton {
+                    client_id: 1234,
+                    context_id: 4321,
+                }),
+            })
+            .unwrap();
+
+        System::current().stop();
+        system.run();
+        let accountant_recording = accountant_recording_arc.lock().unwrap();
+        assert_eq!(accountant_recording.len(), 2);
```

Comment:
If you can't check the whole vector at once, try checking the length of it after you check the contents.